prime-router/docs/features/0005-managing-translation-schemas.md

```diff
@@ -0,0 +1,765 @@
+
+#### Possible Implementations
+- Store a fully resolved translation schema for each sender/receiver
+    - Table for sender and receiver transforms
+    - Store common elements
+    - On updates resolve the translation schemas again and store new versions
+    - Use the same versioning logic as `SETTING` table
+- Update the translation schema definitions to reference database IDs and use those when resolving translation schemas
+    - Rather than `schema: ../common/patient-contact` use `schema: {SOME-DB-ID}`
+- Use a JSONB store
+    - Do away with using YAML as the format and store all the translation schemas as JSON
+    - Shared elements could either be duplicated across translation schemas (store the fully resolved translation schemas) or a sender/receiver translation schemas could be kept separate from the common elements
+
+#### Pros
+- Could potentially enable reporting on what is set in the translation schemas
+- Common elements used across many senders/receivers might be cleaner to implement in a relational database
+- There is an existing pattern that could be used from `SETTING`
+- Reading values from the DB would perform well
+
+#### Cons
+- Would likely be a large re-write as the existing file model does not map well to a relational DB
+- A custom solution would need to be implemented for the HL7->FHIR since the library only reads the mapping from disk
+- Testing this change would be extremely complicated especially considering the current infrastructure around testing the DB
+- Would make it much more difficult to extract libraries for other users to consume as the libraries would be more closely tied to the db
+
+### Azure blob storage (Recommended)
+This solution would maintain the majority of the existing functionality and just provide updates to use the azure blob storage as the underlying "file" storage rather than disk.  This solution would get versioning automatically by enabling [azure blob versioning](https://learn.microsoft.com/en-us/azure/storage/blobs/versioning-overview); this functionality enables automatically creating new versions when uploading a new file and then APIs for rolling back file updates in the event that there was a problem.
+
+Additionally, the versioning has the added flexibility of referencing a specific version of a blob so translation schemas could reference version 3 while the next version is iterated on.
+`https://<storage-account>.blob.core.windows.net/<container>/<blob-name>?versionid=<version-id>` . If the decision is made to continue to store some of the common translation schemas in source code this is an approach that could be adopted here as well by adding a version to the directory paths `file:///{BASE_DIR}/metadata/hl7_mapping/common/patient/v1/patient.yml``
```

Comment:
Regarding "If the decision is made..." I think we should make these decisions now and have the document reflect them, otherwise it just gets kind of burdensome.

Reply:
My opinion is we should go with the idea that "core" mappings/translations will stay in the repo and only "end user" translations will be stored in the blob store as these are the ones that often see the most tweaks and changes. Once we create mappings for an ORU-R01 message, for example, those ideally should never change...

Reply:
My thought was to keep it open ended and aim for a solution that supports all above.  For example the ORU-R01 mapping might change if the RADxMARs spec change.

Reply:
If the RADxMARs spec changes, I think that should be handled by a RADxMARs transform and keep ORU-R01 un-touched.

Reply:
Yep, my comment was more oriented around that the RADxMARS mappings for ORU_01 might change (since they live in the same directory)